apps/reimbursement/contracts/Reimbursement.sol

```diff
@@ -0,0 +1,119 @@
+pragma solidity ^0.4.24;
+
+import "@aragon/os/contracts/apps/AragonApp.sol";
+import "@aragon/os/contracts/kernel/IKernel.sol";
+
+contract Reimbursement is AragonApp {
+  bytes32 public constant ADD_REIMBURSEMENT_ROLE = keccak256("ADD_REIMBURSEMENT_ROLE");
+  bytes32 public constant VETO_REIMBURSEMENT_ROLE = keccak256("VETO_REIMBURSEMENT_ROLE");
+
+  bytes32 public constant KERNEL_APP_ADDR_NAMESPACE = 0xd6f028ca0e8edb4a8c9757ca4fdccab25fa1e0317da1188108f7d2dee14902fb;
+
+  // ensure alphabetic order
+  enum Apps { Contribution, Contributor, Proposal, Reimbursement, Token }
+  bytes32[5] public appIds;
+
+  struct ReimbursementData {
+    address recipient;
+    uint256 amount;
+    address token;
+    bool claimed;
+    bytes32 hashDigest;
+    uint8 hashFunction;
+    uint8 hashSize;
+    uint256 confirmedAtBlock;
+    bool vetoed;
+    bool exists;
+  }
+
+  mapping(uint32 => ReimbursementData) public reimbursements;
+  uint32 public reimbursementsCount;
+
+  uint32 public blocksToWait;
+
+  event ReimbursementAdded(uint32 id, address indexed addedByAccont, uint256 amount);
+  event ReimbursementClaimed(uint32 id, uint256 amount);
+  event ReimbursementVetoed(uint32 id, address vetoedByAccount);
+
+  function initialize(bytes32[5] _appIds) public onlyInit {
+    appIds = _appIds;
+    blocksToWait = 40320; // 7 days; 15 seconds block time
+    initialized();
+  }
+
+  function getContract(uint8 appId) public view returns (address) {
+    IKernel k = IKernel(kernel());
+    return k.getApp(KERNEL_APP_ADDR_NAMESPACE, appIds[appId]);
+  }
+
+  function totalAmount(bool confirmedOnly) public view returns (uint256 amount) {
+    for (uint32 i = 1; i <= reimbursementsCount; i++) {
+      ReimbursementData memory r = reimbursements[i];
+      if (!r.vetoed && (block.number >= r.confirmedAtBlock || !confirmedOnly)) {
+        amount += r.amount; // should use safemath
+      }
+    }
+  }
+
+  function getReimbursement(uint32 reimbursementId) public view returns (uint32 id, address recipient, uint256 amount, address token, bool claimed, bytes32 hashDigest, uint8 hashFunction, uint8 hashSize, uint256 confirmedAtBlock, bool exists, bool vetoed) {
```

Comment:
Maybe we should just call this `get`, analog to `add` below?

Reply:
yeah, thought about that, too. but we had `getContribution` in the contribution contract. 
do you think that's the best pattern? :+1:

Reply:
Yeah, I think it should be consistent, and `contribution.add` is nicer for clients than `contribution.addContribution`. May as well imply the scope for all functions then.